### DISCUSSION TITLE
Hiding also H2, H3 in Markdown from navigation not only H1

### DISCUSSION THREAD

**Discussion by simondueckert** [squidfunk/mkdocs-material#7630] [Q&A]
I use mkdocs material to create a number of guidelines. When the title in the navigation is the same like a level 1 heading in the markdown file the title is only shown once. If I have level 2 or 3 headings in markdown the title is shown twice (see screenshot). Is there any way to suppress the display of terms from navigation so only headings from the markdown files are shown?

<img width="1397" alt="Bildschirmfoto 2024-10-19 um 12 41 58" src="https://github.com/user-attachments/assets/62135915-1c7b-47e4-88fb-2bbbc4b8f3cf">

**Comment by alexvoss** [comment]
You should really have a level one heading int the Markdown. Is the reason you have a level 2 heading that you want the formatting to be different? If so, then use CSS to change this instead of skipping levels in the structure of your pages. If I guessed wrong then please le us know what it is you really want to achieve so we can advise.

**Reply by simondueckert** [reply]
I have level 2 and level 3 headings in the markdown because I also generate a PDF out of the markdown (with pandoc) and need an hierarchic table of content there. Do you have a tip for me where I have to put the CSS? Is it the stylesheets/extra.css I already use?

**Reply by alexvoss** [reply]
Yes, that would be the place. Are you using Pandoc for the PDF by any chance? It might be an idea to see if the adjustment could be made on the route to the PDF instead of using the wrong levels in the Markdown. One reason why this seems the better path to me is accessibility.  **Edit:** was thinking of the [`--shift-heading-level-by`](https://pandoc.org/MANUAL.html#reader-options) option.

**Reply by kamilkrzyskow** [reply]
You can use theme extension to change the `partials/content.html` to remove the hacky H1 creation:
- https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme
- https://github.com/squidfunk/mkdocs-material/blob/ade227cbbb2a2debcf7c56cc2426696029e74978/src/templates/partials/content.html#L31-L37

EDIT: Or if you don't want the H1 tag only when the H2 matches the page's title you can modify it like this:
```jinja2
{% if "\x3ch1" not in page.content and page.toc.items[0].title != page.title %}
  <h1>{{ page.title | d(config.site_name, true)}}</h1>
{% endif %}
```

EDIT2: But when it comes to accessibility a proper hierarchy tree of the headings could be important, so a lack of H1 could be a mistake, like Alex is saying. Depending on the used PDF creation method, you could perhaps have 2 different workflows for creating the PDF and deploying for the web.
